### PR TITLE
Add initial Pilot monitoring dashboard to grafana

### PR DESCRIPTION
**Pull request by douglas-reid**
This PR adds an initial monitoring dashboard to the existing Istio grafana add-on for k8s. The dashboard provides a high-level overview of resource consumption, pilot-generated metrics, and xDS cluster stats from envoy.

**Reviewers**: As I'm not intimately familiar with running Pilot, this PR represents a first stab at getting something to play with as we push towards towards an "alpha" status dashboard for use in Istio. I'm very open to suggestions as to ways that this can be improved.

However, I feel like this might require playing with the dashboard a bit to get a feel for how it performs and see what is missing / not working as expected (for instance, it appears that cache size information is not being calculated on the pilot side correctly). So, it might be beneficial to take an initial baseline and iterate through experience. I'll leave that to your discretion in review.

A subsequent PR will address (forcing) the update the addon yamls so that this will be easier to play with via `kubectl apply -f install/kubernetes/addons/grafana.yaml` (right now requires manual update).

pilot-dashboard

### CODE REVIEW

**Comment by kyessenov** [comment]
Very cool!

**Comment by rshriram** [comment]
This is awesome!
do you think separating hits and misses or having one per cds, rds, lds, eds makes sense?

File: mixer/deploy/kube/conf/Dockerfile [hunk 1]
```diff
@@ -13,6 +13,7 @@ ENV GF_AUTH_ANONYMOUS_ORG_ROLE=Admin
 COPY ./start.sh /start.sh
 COPY ./grafana-dashboard.json /grafana-dashboard.json
 COPY ./mixer-dashboard.json /mixer-dashboard.json
+COPY ./pilot-dashboard.json /pilot-dashboard.json
```
Comment (mattdelco) [inline mixer/deploy/kube/conf/Dockerfile]: This file should be added to the GRAFANA_FILES list in the top-level Makefile.

**Comment by douglas-reid** [comment]
@rshriram I thought about separating out the xDS's into separate rows might be useful. I can go ahead and do that now.

**Comment by douglas-reid** [comment]
Updated dashboard and added files to Makefile. PTAL.

pilot-dashboard

**Comment by douglas-reid** [comment]
/assign zackbutcher

zack, any chance you can give me a "/approve" ?

**Comment by ZackButcher** [comment]
Sorry this slipped by, still getting new email filters in place with the larger volume of GH reviews due to CODEOWNERS changes.

/approve

**Pull request merged**